src/appointment/appointment.service.ts

```diff
@@ -0,0 +1,94 @@
+import { Injectable, HttpException, HttpStatus } from '@nestjs/common';
+import { InjectModel } from 'nestjs-typegoose';
+import { Appointment } from 'src/model/appointment.model';
+import {AppointmentStatus} from 'src/enum/appointment.enum'
+import { ReturnModelType } from '@typegoose/typegoose';
+import { AppointmentDTO, EditAppointmentDTO } from './appointment.dto';
+import { UserService } from 'src/user/user.service';
+import { UserRole } from 'src/enum/user.enum';
+
+@Injectable()
+export class AppointmentService {
+    constructor(
+        @InjectModel(Appointment)
+        private readonly model: ReturnModelType<typeof Appointment>,
+        private readonly userService: UserService,
+    ) {}
+
+    async createAppointment(
+        appointment: AppointmentDTO,
+        studentId: string,
+    ): Promise<Appointment> {
+        const student = await this.userService.findById(studentId);
+        const tutor = await this.userService.findById(appointment.tutorId);
+        const appointmentObject = new this.model({
+            startTime: appointment.startTime,
+            endTime: appointment.endTime,
+            location: appointment.location,
+            price: appointment.price,
+            student,
+            tutor,
+        });
+        if(tutor.role===UserRole.Tutor) return appointmentObject.save();
+        throw new HttpException('invalid tutor id',HttpStatus.NOT_ACCEPTABLE);
```

Comment:
- Capitalize `I` e.g. 'Invalid ...' since it is the begining of the sentence

- Consider using `if(tutor.role!==UserRole.Tutor)` to check validity first and `throw` immediately if it's not valid. So it does not have to create unused `model` in case that the request should be denied.